src/reporter.cc

```diff
@@ -156,6 +156,253 @@ Napi::Value send_span(const Napi::CallbackInfo& info, send_generic_span_t send_f
 
 }
 
+enum SMFlags {
+  kSMFlagsTesting = 1 << 0,
+  kSMFlagsNoop = 1 << 1
+};
+//
+// internal function used by sendMetric() (deprecated) and sendMetrics().
+//
+Napi::Value send_metrics_core (Napi::Env env, Napi::Array metrics, uint64_t flags) {
+  int64_t goodCount = 0;
+  const char* service_name = "";
+  bool testing = flags & kSMFlagsTesting;
+  bool noop = flags & kSMFlagsNoop;
+
+  Napi::Array errors = Napi::Array::New(env);
+  Napi::Array echo;
+  Napi::Object echoTags;
+  if (testing) {
+    echo = Napi::Array::New(env);
+  }
+
+  //
+  // loop through each metric and send if valid else
+  // report back in errors array.
+  //
+  for (size_t i = 0; i < metrics.Length(); i++) {
+    bool is_summary = false;
+    std::string name;
+    int64_t count;
+    double value = 0;
+    bool add_host_tag = false;
+
+    Napi::Value element = metrics[i];
+
+    // little lambda for errors.
+    auto set_error = [&](const char* code) {
+      Napi::Object err = Napi::Object::New(env);
+      err.Set("code", code);
+      err.Set("metric", element);
+      errors[errors.Length()] = err;
+    };
```

Comment:
I haven't seen this kind of init before. Is this equivalent to an anonymous function?

Reply:
not sure what you're asking about here.

Reply:
if you're asking about the set_error() function - it's a lambda function. the main benefit to it is that, by adding the `&` like `[&]` it can reference local variables without needing to pass them as arguments.

Reply:
:+1: